File: MessagingApp-App/ClientApp/src/app/_models/messageReaction.ts
import { Message } from "./message";
import { Reaction } from "./reaction";
import { User } from "./user";

export interface MessageReaction {
  userId?: number;
  user: User;
  messageId?: number;
  message?: Message;
  reaction?: Reaction;
  created?: Date;
  lastUpdated?: Date;
}

File: MessagingApp-ApiContext/Models/MessageReaction.cs
using System;

namespace MessagingApp.ApiContext.Models
{
    public class MessageReaction
    {
        public int UserId { get; set; }
        public User User { get; set; }
        
        public int MessageId { get; set; }
        public Message Message { get; set; }

        public Reaction Reaction { get; set; }

        public DateTime Created { get; set; }

        public DateTime LastUpdated { get; set; }
    }
}
